src/forward_link.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+
+def mask_values(df: pd.DataFrame, target_variable: str, expr: str) -> pd.Series:
+    """Convert values in a dataframe column to 0 based on a python expression
+
+    Parameters
+    ----------
+    df : pd.Dataframe
+        Pandas dataframe of original data.
+    target_variable : List[str] or str
+        Column name(s) containing target variable(s).
+    query : str
+        The expression to evaluate, see here:
+        https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.eval.html
+
+    Returns
+    -------
+    df : pd.Series
+
+
+    """
+    masked_column = df[target_variable].copy()
+
+    try:
+        masked_column.loc[~(df.eval(expr))] = np.nan
+
+    except ValueError:
+        print(
+            f"""{expr} is not a valid expression,
+        the code uses ~(df.eval({expr}) to mask the dataframe, please see here:
+        https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.eval.html
+        """
+        )
+
+    return masked_column
+
+
+def calculate_imputation_link(
+    df: pd.DataFrame,
+    groups: List[str] or str,
```

Comment:
Only minor point to think about:
Is it better to explicitly give group and period as separate inputs.
It's more elegant to use list as input, but it might be slightly confusing if other functions require separate inputs